audformat/core/utils.py

```diff
@@ -568,24 +568,11 @@ def intersect(
 
         _convert_single_level_multi_index(objs)
 
-        index = objs[0]
-        for obj in objs[1:]:
-            index = index.intersection(obj)
-
-        # index.intersection() does not preserve string dtype
-        # for MultiIndex
-        if isinstance(index, pd.MultiIndex):
-            obj = objs[0]
-            dtypes = {
-                name: dtype for name, dtype in zip(obj.names, obj.dtypes)
-                if dtype == 'string'
-            }
-            index = set_index_dtypes(index, dtypes)
-
-    # We use len() here as index.empty takes a very long time
-    if len(index) == 0 and is_segmented_index(index):
-        # asserts that start and end are of type 'timedelta64[ns]'
-        index = segmented_index()
+        index = union(objs)
+        mask = np.ones(len(index), dtype=bool)
+        for obj in objs:
+            mask &= index.isin(obj)
```

Comment:
yes

Reply:
Great, even simpler to understand then before.

The only downside is that it is slightly slower. I tried it on some large mixed tables from `salamander` and the execution time increased from 2.3 s to 4.5 s.

But the good news is, it can be reduced to at least 2.8 s.
There is no need to take the union, instead you could start with the smallest index:
```python
# start intersection from smallest index
index = min(objs, key=len)
```
If you do so we haver two failing tests as `Int64Index()` is returned instead of `Index` with dtype 'Int64', but those should be identical.

Another solution for speedup could be to check after each iteration if any intersection is left, and if not break the for loop and return an empty index.

Reply:
> Another solution for speedup could be to check after each iteration if any intersection is left, and if not break the for loop and return an empty index.

I now sort the objects by length and break early if no intersection is possible. Can you please check if this helps?

Reply:
But ok, it can of course only help in cases where there is no intersection.

Reply:
It helps, it is now down to 2.6 s, which should be fine.

Reply:
But you are still relying on `union()`, is this really needed?

Reply:
I now also implemented your first (very clever!) solution. I slightly changed the tests to avoid the failing example. But I think it even makes more sense now, because the intersection of a `int64` and `Int64` index can never include `nan`, so there is no need to convert anyway.

Reply:
Best part: after I also adapted the first of the docstring examples it was suddenly returning `'0 days'` for the later example again :D

Reply:
Very cool, we are now down to 1.0 s, so even faster than before :)